service/controller/clusterapi/v17/key/cluster_provider.go

```diff
@@ -60,3 +60,18 @@ func g8sClusterStatusFromCMAClusterStatus(cmaStatus *runtime.RawExtension) g8sv1
 
 	return g8sStatus
 }
+
+func setG8sClusterStatusToCMAClusterStatus(cluster cmav1alpha1.Cluster, status g8sv1alpha1.AWSClusterStatus) cmav1alpha1.Cluster {
+	var err error
+
+	if cluster.Status.ProviderStatus == nil {
+		cluster.Status.ProviderStatus = &runtime.RawExtension{}
+	}
+
+	cluster.Status.ProviderStatus.Raw, err = json.Marshal(status)
+	if err != nil {
+		panic(err)
```

Comment:
Why are we panic-ing here? Is there a reason why we can't just propagate the error?

Reply:
For the same reason as in other key functions that deal with provider extension. It makes use of key functions very cumbersome if there would need to be error handling. If that json marshalling fails, there's something pretty seriously wrong anyway.